src/main/kotlin/com/devfloor/untitled/article/application/ArticleFacade.kt

```diff
@@ -36,16 +34,31 @@ class ArticleFacade(
 
     @Transactional(readOnly = true)
     fun showByArticleId(articleId: Long): ArticleResponse {
-        val article = articleService.showById(articleId)
-        val options = articleOptionService.showAllByArticle(article)
-        val hashtags = articleHashtagService.showAllByArticle(article)
-        val favorites = articleFavoriteService.showAllByArticle(article)
+        articleService.showById(articleId).let {
+            val options = articleOptionService.showAllByArticle(it)
+            val hashtags = articleHashtagService.showAllByArticle(it)
+            val favorites = articleFavoriteService.showAllByArticle(it)
+
+            return ArticleResponse(
+                options = options,
+                article = it,
+                hashtags = hashtags,
+                favorites = favorites,
+            )
+        }
```

Comment:
일단 `let` 앞에서 한 번 개행해주는게 더 좋을 것 같아.

그리고 이것도 개인적인 제안이긴 한데 let 메서드의 인자는 T -> R 로 변환해주는 함수를 받거든. 그런데 let 내부에서 return을 사용하는 게 조금 이상한 것 같아. let 메서드를 잘못 쓴 것 같은 느낌이야.

만약 저런 식으로 작성하는 거라면 
```suggestion
        return articleService.showById(articleId)
            .let {
                val options = articleOptionService.showAllByArticle(it)
                val hashtags = articleHashtagService.showAllByArticle(it)
                val favorites = articleFavoriteService.showAllByArticle(it)

                ArticleResponse(
                    options = options,
                    article = it,
                    hashtags = hashtags,
                    favorites = favorites,
                )
            }
```

이런 식으로 쓰는 게 좀 더 맞을 것 같아. 그래서 이러면 오히려 로직이 더 길어지는 것 같아서.. 처음 작성한 코드도 충분히 깔끔한 것 같은데 한 번 참고해주면 좋을 것 같아!

Reply:
추가적으로 체이닝 메서드에서의 개행 시점(?)은 아래 컨벤션 참고해보면 좋을 것 같아!
https://kotlinlang.org/docs/coding-conventions.html#wrap-chained-calls

Reply:
아 이 부분에 일단 체이닝으로 한 이유는, Kotlin적으로 관용적으로 사용하라고 한것중 하나가 scope function이라고 적혀있더라고.
내 생각엔 scope function을 사용하면 좋을만한 부분에는 사용하는게 좋다고 생각했고, 이 부분도 article에 대한 조회가 이루어지고 모든 부분이 수행되어야하니까 이렇게 묶으면 좋지 않을까 생각했었어.

그런데 여기서 체이닝 메서드를 개행하지 않은 것은 내가 컨벤션중 반영하지 않았던 부분이야.! 한줄짜리 간단한 코드 에서도(e.g. `a.let{ doTask() }`) 항상 개행을하면 좋지 않을거같았거든. 복잡한 체이닝이 되어있다면 개행을 하는게 좋을거같은데 컨벤션에도 써있듯 괜찮을 것 같아서 개행을 생략한거긴 해 ! 뭐 근데 둘다 좋을거같아서 개행해도 상관없긴할거같은데, 내 생각엔 체이닝이 두줄 이상 수행될 때 개행하면 더 좋지 않을까?! 이런 생각이었어 !

Reply:
위에 말한 것 처럼 `a.let{ doTask() }` 같은 경우 개행하는게 오히려 별로일 것 같은데 지금은 이미 `articleService.showById(articleId)` 로 메서드를 호출한 상태라 다음 라인으로 넘기는 것도 가독성에서 좋을 것 같아.